Subject: [PATCH] Optimize AbsListView to reduce click operation latency

Separate performClick from delayed message. Then we can do
performClick in advance of PRESSED_STATE_DURATION(64ms) delay.
This will benefit for click operation latency. The delayed
message is still used to clear pressed status.

diff --git a/core/java/android/widget/AbsListView.java b/core/java/android/widget/AbsListView.java
index a733b44ddba..d9921388d8e 100644
--- a/core/java/android/widget/AbsListView.java
+++ b/core/java/android/widget/AbsListView.java
@@ -4066,6 +4066,11 @@ private void onTouchUp(MotionEvent ev) {
                                 }
                                 mSelector.setHotspot(x, ev.getY());
                             }
+                            if (!mDataChanged && !mIsDetaching && isAttachedToWindow()) {
+                                if (!post(performClick)) {
+                                    performClick.run();
+                                }
+                            }
                             if (mTouchModeReset != null) {
                                 removeCallbacks(mTouchModeReset);
                             }
@@ -4076,9 +4081,6 @@ public void run() {
                                     mTouchMode = TOUCH_MODE_REST;
                                     child.setPressed(false);
                                     setPressed(false);
-                                    if (!mDataChanged && !mIsDetaching && isAttachedToWindow()) {
-                                        performClick.run();
-                                    }
                                 }
                             };
                             postDelayed(mTouchModeReset,
